### DISCUSSION TITLE
Asrock z370 extreme 4 Secure check fail headache

### DISCUSSION THREAD

**Discussion by BulletWilly** [terminatorul/NvStrapsReBar#54] [Q&A] [answered]
Hi

I'm having a lot of difficulties flashing the modded UEFI with Asrock Instant Flash

- Followed all prior steps with the UEFI tool just fine, inserted after at the end of the correct volume on clean current bios.

- Secure check failed as expected.

- Tried with UEFI patch steps too

- Tried Flashrom via USB but failed

- UBU to disable/remove Asrocks secure check which I've recently found out doesn't work on Intel 300+ chipsets.

- Intel flash programming tool which gave:

Error 201: [FPTw64.exe] cannot be run on the current platform.
Please contact your vendor.

I'm struggling with where to go from here, looked at similar posts on lvl1techs etc.

Any help would be greatly appreciated, I'm starting to think that without hardware mods I can't flash this.
 
Asrock z370 extreme 4 Bios 4.21B (Has 4G decoding and C.A.M/Clever Access Memory)
8700k
MSi Gaming X Trio 2080 Super

**Comment by BJTorgs** [comment]
Is it AMI bios by chance?

**Reply by BulletWilly** [reply]
Yeah, AMI aptio 5 I believe

**Reply by BJTorgs** [reply]
You can try this guide. There's an Asrock specific section in the guide.
https://winraid.level1techs.com/t/guide-how-to-flash-a-modded-ami-uefi-bios/30627

**Reply by BulletWilly** [reply, marked as answer]
Yeah, that was the guide I followed originally but nothing worked on the Asrock section, thanks for the advice though.

Nonetheless, I've managed to fix it now, I realised I had to use the exact version of CSME tools/intel flash programming toolset for my system

Also, I had to disable Windows 11 driver exclusion for the tool to work

For me that was v11 
![image](https://github.com/terminatorul/NvStrapsReBar/assets/131589179/eeef281b-49c8-4784-8056-82e71bc26fa9)

Download here in section C https://winraid.level1techs.com/t/intel-converged-security-management-engine-drivers-firmware-and-tools-2-15/30719?page=89 

Smooth as butter after that, added ffs module using uefi tool, didn't need uefi patch and flashed.

After setting with rebarstraps exe and testing it in game with dead space remake it gave me just over **30 FPS UPLIFT**!! overall and 1% lows

**Comment by ags1234** [comment]
Does anyone have a good, modified recent BIOS for this board?

**Reply by Koekieezz** [reply]
@ags1234 do you still need it?

**Reply by ags1234** [reply]
No, I changed the board since then. Thank you.
